backend/src/services/lectures/lectures.repo.js

```diff
@@ -112,6 +173,44 @@ class LecturesRepo {
     });
   }
 
+  /**
+   * Activate attendance for a lecture by generating a code and starting the 5-minute timer.
+   * This should be called when the professor/TA clicks "Start Attendance" button.
+   *
+   * @param {number} lectureId - ID of the lecture
+   * @param {number} courseId - ID of the course (for generating unique code)
+   * @returns {Promise<Object>} Updated lecture object with code and expiration timestamps
+   */
+  async activateAttendance(lectureId, courseId) {
+    const code = await this.generateUniqueCode(courseId);
+    const codeGeneratedAt = new Date();
+    const codeExpiresAt = new Date(codeGeneratedAt.getTime() + 5 * 60 * 1000); // 5 minutes
+
+    return this.db.lectures.update({
+      where: { id: lectureId },
```

Comment:
[Suggestion]
We should also condition on the course id for safety. If not, I could provide a valid lecture ID that is not in the course that I am in and successfully change that lecture's data.